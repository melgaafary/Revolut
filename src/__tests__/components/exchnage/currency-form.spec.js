import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import CurrencyForm from '../../../components/exchange/currency-form';

describe('[Component] CurrencyForm', () => {
  const mockOnChange = jest.fn();
  it('should match the snapshot', () => {
    const component = <CurrencyForm name="inGoing" onChange={mockOnChange} />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should show correct ingoing value', () => {
    const component = mount(
      <CurrencyForm value={200} name="inGoing" onChange={mockOnChange} />,
    );
    expect(component.find('input[type="number"]').props().value).toEqual(200);
    expect(component.find('span').text()).toEqual('-');
  });
  it('should show correct outgoing value', () => {
    const component = mount(
      <CurrencyForm value={200} name="outGoing" onChange={mockOnChange} />,
    );
    expect(component.find('input[type="number"]').props().value).toEqual(200);
    expect(component.find('span').text()).toEqual('+');
  });
  it('should show error', () => {
    const component = mount(
      <CurrencyForm
        value={200}
        name="outGoing"
        onChange={mockOnChange}
        error="Exceeds limit"
      />,
    );
    expect(
      component
        .find('span')
        .last()
        .text(),
    ).toEqual('Exceeds limit');
  });
});
