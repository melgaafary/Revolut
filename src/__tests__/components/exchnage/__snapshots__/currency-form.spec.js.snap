// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] CurrencyForm should match the snapshot 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 cfvjOn"
>
  <div
    className="StyledBox-sc-13pk1d4-0 eitMOG"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 irGvOS"
    >
      
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 jJdKAu"
    >
      <input
        className="sc-AxjAm kcUFxv"
        name="inGoing"
        onChange={[MockFunction]}
        placeholder="0"
        type="number"
        value=""
      />
    </div>
  </div>
</div>
`;
